client/components/data/query-wordads-settings/index.jsx

```diff
@@ -2,19 +2,32 @@
  * External dependencies
  */
 import React from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestWordadsSettings } from 'calypso/state/wordads/settings/actions';
+import isJetpackSite from 'calypso/state/sites/selectors/is-jetpack-site';
+import { requestJetpackSettings } from 'calypso/state/jetpack/settings/actions';
+import isRequestingJetpackSettings from 'calypso/state/selectors/is-requesting-jetpack-settings';
 
 export default function QueryWordadsSettings( { siteId } ) {
 	const dispatch = useDispatch();
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isFetchingSettings = useSelector( ( state ) =>
+		isRequestingJetpackSettings( state, siteId, null )
+	);
 
 	React.useEffect( () => {
 		dispatch( requestWordadsSettings( siteId ) );
-	}, [ dispatch, siteId ] );
+
+		// WordAds settings on Jetpack sites are not available on the WordAds API endpoint, so we
```

Comment:
Just double checking here.

Making changes to `/earn/ads-settings/` does reflect to `https://public-api.wordpress.com/rest/v1.1/sites/[SITE_ID]/wordads/settings`, are we sure we need to use `requestJetpackSettings`?

Reply:
And if we indeed need `requestJetpackSettings` does it mutually exclude needing `requestWordadsSettings` for Atomic sites?

Reply:
> Making changes to /earn/ads-settings/ (for an Atomic site) does reflect to https://public-api.wordpress.com/rest/v1.1/sites/[SITE_ID]/wordads/settings,

Not always. If you update the data directly on the remote site (e.g. by going to `/wp-admin/admin.php?page=jetpack#/traffic` and making some changes there), these changes won't be reflected in the WordAds endpoint response.

That's why this PR always uses the data from the Jetpack settings endpoint, to make sure that the data is up-to-date.

> And if we indeed need requestJetpackSettings does it mutually exclude needing requestWordadsSettings for Atomic sites?

No because there are some settings which are processed exclusively by the WordAds endpoint and not by the Jetpack settings endpoint (e.g. the PayPal email address).